File: LatihanInheritanceC++/Kendaraan/Mobil.cpp
#include "Mobil.hpp"

Mobil ::Mobil() : Kendaraan()
{
    this->supir = "XXXX";
}
// user-defined ctor
Mobil ::Mobil(string supir, int nomorKendaraan, string merk, int tahunKeluaran) : Kendaraan(nomorKendaraan, merk, tahunKeluaran)
{
    this->supir = supir;
}
// cctor
Mobil ::Mobil(const Mobil &m) : Kendaraan(m)
{
    this->supir = m.supir;
}
// dtor: tidak diperlukan karena mengalokasikan memori heap
Mobil ::~Mobil(){};

void Mobil ::PrintInfo()
{
    Kendaraan::PrintInfo();
    cout << "Kategori: Mobil" << endl;
    cout << "Supir: " << this->supir << endl;
}

int Mobil ::BiayaSewa(int lamaSewa)
{
    return 500000 * lamaSewa;
}

File: LatihanInheritanceJava/src/Mobil.java
class Mobil extends Kendaraan {
    private String supir;

    public Mobil() {
        super();
        this.supir = "XXXX";
    }

    public Mobil(int nomorKendaraan, String merk, int tahunKeluaran, String supir) {
        super(nomorKendaraan, merk, tahunKeluaran);
        this.supir = supir;
    }

    public Mobil(Mobil Mb) {
        super(Mb);
        this.supir = Mb.supir;
    }

    public long BiayaSewa(int lamaSewa) {
        return 500000 * lamaSewa;
    }

    public void printInfo() {
        super.printInfo();
        System.out.printf("Kategori: %s\n", "Mobil");
    }
}